clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/DeployAppengineAtomicOperation.groovy

```diff
@@ -171,6 +206,16 @@ class DeployAppengineAtomicOperation implements AtomicOperation<DeploymentResult
     return versionName
   }
 
+  static List<String> updateConfigRuntime(List<String> configFiles) {
+    if (!configFiles) {
+      return []
+    } else {
+      return configFiles.collect { configFile ->
+        return configFile.replaceAll(/runtime\s*\:\s*[a-zA-Z]+/, "runtime: custom")
```

Comment:
Shouldn't the user be responsible for submitting the correct runtimes?

Reply:
`gcloud app deploy` verifies that source code exists in the current directory and validates that it matches the app.yaml's runtime.  This validation runs even when deploying from an image URL, which is odd since in this case the local source code is never actually used for anything.  This seemed like the easiest workaround for gcloud's slightly errant behaviour.

With all that said, it doesn't really answer your question.  I guess we could allow the user to submit whatever runtime they want but could warn them that any runtime except custom will fail.  Or we could disallow form submission with any runtime other than custom.

Reply:
I would lean towards failing with incorrect runtimes - my (limited) understanding is that writing a custom runtime requires extra work, and switching it for a user without them knowing might cause strange breakages.

Reply:
`runtime: custom` is essentially a confirmation to gcloud that the user wants to deploy their own container. The other runtimes are used when gcloud builds a new container - which is the old codepath, not this one. i.e., if the user supplied a container image, they definitely want `runtime: custom`.

Reply:
(Though I think Scott is removing this anyway)